aim/web/api/runs/object_views.py

```diff
@@ -95,6 +95,38 @@ async def sequence_batch_api(run_id: str,
             def blobs_batch_api(uri_batch: URIBatchIn):
                 return StreamingResponse(get_blobs_batch(uri_batch, get_project_repo()))
 
+        # run sequence batch API
+        step_of_sequence_endpoint = f'/{{run_id}}/{seq_name}/get-step/'
+
+        @router.post(step_of_sequence_endpoint, response_model=List[ObjectSequenceBaseView])
+        async def step_of_sequence(run_id: str,
+                                   requested_traces: RunTracesBatchApiIn,
+                                   index_range: Optional[str] = '', index_density: Optional[int] = 5,
+                                   record_step: int = None):
+
+            index_range = checked_range(index_range)
+            CustomObjectApiConfig.check_density(index_density)
+
+            repo = get_project_repo()
+            run = repo.get_run(run_id)
+            if not run:
+                raise HTTPException(status_code=404)
+
+            api = CustomObjectApi(seq_name, resolve_blobs=cls.resolve_blobs)
+            api.set_dump_data_fn(cls.dump_record_fn)
+            api.set_requested_traces(run, requested_traces)
+
+            if record_step is None:
+                total_record_range = api.get_total_record_range()
```

Comment:
total range will be calculated twice. 1st time here, 2nd time after `set_ranges` call. better to leave this logic in `CustomObjectApi` and do the calculation the only once